components/suggestion/switch_channel_provider.jsx

```diff
@@ -524,4 +524,42 @@ export default class SwitchChannelProvider extends Provider {
             component: ConnectedSwitchChannelSuggestion,
         });
     }
+
+    formatAllRecentChannelsAndDispatch(resultsCallback) {
+        const recentChannels = getAllRecentChannels(getState());
+
+        const channels = [];
+        for (let i = 0; i < recentChannels.length; i++) {
+            const channel = recentChannels[i];
+
+            let wrappedChannel = {channel, name: channel.name, deactivated: false};
+            if (channel.type === Constants.GM_CHANNEL) {
+                wrappedChannel.name = channel.display_name;
+            } else if (channel.type === Constants.DM_CHANNEL) {
+                const user = getUser(getState(), Utils.getUserIdFromChannelId(channel.name));
+
+                if (!user) {
+                    continue;
+                }
+
+                wrappedChannel = this.userWrappedChannel(
+                    user,
+                    channel,
+                );
+            }
+            wrappedChannel.type = Constants.MENTION_RECENT_CHANNELS;
+            channels.push(wrappedChannel);
+        }
+
+        const channelNames = channels.map((wrappedChannel) => wrappedChannel.channel.id);
```

Comment:
hmm why is this `channel.id` if we're returning `channelNames`? does it get resolved somewhere?

Reply:
I wrote this using the seeing the above function of `formatUnreadChannelsAndDispatch` and I have not looked why it is written that way.I will check it.